proposals/jvm-meta-annotations-artifact.md

```diff
@@ -0,0 +1,160 @@
+It might look like:
+```kotlin
+enum class ApplicabilityKind {
+    RETURN_TYPE, VALUE_PARAMETER, FIELD, TYPE_USE, TYPE_ARGUMENT
+}
+```
+
+All elements should work just the same as relevant entries from `ElementType` do for `@TypeQualifierDefault`,
+beside `TYPE_ARGUMENT`.
+The latter one should have the following effect: if a default qualifier is determined
+to be applied to some top-level type (using the same logic as for `@TypeQualifierDefault`)
+and the set of applicability kinds contain `TYPE_ARGUMENT` then this qualifier should also be applied
+to all of it's type arguments (recursively).
+
+#### Parameter in @Alias (Option 1)
+The first option is adding `propagateTo: Array<DefaultApplicabilityType>` parameter to `meta.Alias` annotation.
+
+Thus, declaration of `AllParametersAreNullableByDefault` would look like this:
+```kotlin
+@Alias(Nullable::class, propagateTo=[ApplicabilityKind.VALUE_PARAMETER])
+annotation class AllParametersAreNullableByDefault
+```
+
+*Known pros:* 
+- It's nice to have one annotation for all issues.
+- Probably, for someone such form of default qualifiers would look nicer than one suggested by JSR-305.
+
+*Known cons:*
+- It would look a little weird when it's used in cases like the one below:
+```kotlin
+@Alias(Nullable::class)
+@UnderMigration(...)
+annotation class MyNullable
+
+@Alias(MyNullable::class, propagateTo=[ApplicabilityKind.PARAMETER])
+annotation class MyAllParametersAreNullableByDefault
+```
+- Also it might be confusing when one alias (probably with non-empty `propagateTo`)
+references another alias with non-trivial `propagateTo` argument
+
+#### Separate annotation (Option 2)
+Another option would be a separate annotation `meta.ApplyByDefault` with vararg-parameter
+of type `DefaultApplicabilityType`:
+```kotlin
+@Target(AnnotationTarget.ANNOTATION_CLASS)
+annotation class ApplyByDefault(val qualifier: KClass<*>, vararg val elements: ApplicabilityKind)
+```
+
+Basically, it should work just the same as `@TypeQualifierDefault`, e.g. it might be used like:
+```kotlin
+@ApplyByDefault(MyNullable::class)
+annotation class MyAllParametersAreNullableByDefault
+```
+
+### Avoidance of @UnderMigration annotation
+After revisiting design for meta-annotations it looks like a natural thing to set up
+a migration status for an annotation with its argument instead of another meta-annotation.
+
+Thus, the idea is to add `migrationStatus` parameter to `meta.Alias` 
+(and to `meta.ApplyByDefault` if we decide to introduce the latter) with default value to be `STRICT`.
+
+This argument should be processed by the compiler in the same way as for [`@UnderMigration`](https://github.com/Kotlin/KEEP/blob/master/proposals/jsr-305-custom-nullability-qualifiers.md#undermigration-annotation) annotation 
+having the same argument value.
+
+### Details on artifact
+
+There is already an artifact called [kotlin-annotations-jvm](https://mvnrepository.com/artifact/org.jetbrains.kotlin/kotlin-annotations-jvm)
```

Comment:
Same open question than for the package, isn't Kotlin scope too narrow for such general nullability need?